assets/pages/new-contributions-landing/components/SetPassword/SetPassword.jsx

```diff
@@ -0,0 +1,153 @@
+import { setThankYouPageStage, setPasswordHasBeenSubmitted, updatePassword, type Action } from '../../contributionsLandingActions';
+import { checkEmail, emailRegexPattern } from '../../formValidation';
+import { ButtonWithRightArrow } from '../ButtonWithRightArrow';
+
+// ----- Types ----- //
+
+/* eslint-disable react/no-unused-prop-types */
+type PropTypes = {
+  contributionType: Contrib,
+  email: string,
+  password: string,
+  guestAccountCreationToken: string,
+  setThankYouPageStage: (ThankYouPageStage) => void,
+  setPasswordHasBeenSubmitted: () => void,
+  passwordHasBeenSubmitted: boolean,
+  updatePassword: (Event) => void,
+  csrf: CsrfState,
+};
+/* eslint-enable react/no-unused-prop-types */
+
+
+// ----- State Maps ----- //
+
+const mapStateToProps = state => ({
+  contributionType: state.page.form.contributionType,
+  email: state.page.form.formData.email,
+  password: state.page.form.setPasswordData.password,
+  passwordHasBeenSubmitted: state.page.form.setPasswordData.passwordHasBeenSubmitted,
+  guestAccountCreationToken: state.page.form.guestAccountCreationToken,
+  csrf: state.page.csrf,
+});
+
+function mapDispatchToProps(dispatch: Dispatch<Action>) {
+  return {
+    setThankYouPageStage: (thankYouPageStage: ThankYouPageStage) => {
+      dispatch(setThankYouPageStage(thankYouPageStage));
+    },
+    setPasswordHasBeenSubmitted: () => {
+      dispatch(setPasswordHasBeenSubmitted());
+    },
+    updatePassword: (event: Event) => {
+      if (event.target instanceof HTMLInputElement) {
+        dispatch(updatePassword(event.target.value));
+      }
+    },
+  };
+}
+
+
+// ----- Functions ----- //
+
+function onSubmit(props: PropTypes): Event => void {
+  return (event) => {
+    props.setPasswordHasBeenSubmitted();
+    event.preventDefault();
+
+    if (!(event.target: any).checkValidity()) {
+      return;
+    }
+
+    // TODO: send user to thank you page with error if password was not set
+    setPasswordGuest(props.password, props.guestAccountCreationToken, props.csrf)
+      .then((response) => {
+        if (response === true) {
+          props.setThankYouPageStage('thankYouPasswordSet');
+        } else {
+          props.setThankYouPageStage('thankYouPasswordNotSet');
+        }
+      });
+  };
+}
+
+
+// ----- Render ----- //
+
+function SetPassword(props: PropTypes) {
+  return (
+    <div className="set-password__container">
+      <h1 className="header">Set up a free account to manage your payments</h1>
+      <section className="set-password">
+        <p className="blurb">
+          Thank you for a valuable contribution. As a contributor, being signed in means you will no
+          longer see the “Since you’re here …” messages asking you to support our journalism.
+        </p>
+        <form onSubmit={onSubmit(props)} className={classNameWithModifiers('form', ['contribution'])} noValidate>
+          <NewContributionTextInput
```

Comment:
On the landing page we have `type="email"` and `autocomplete="email"` (the latter giving [more specific guidance](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#attr-autocomplete) to the browser)

I know it's not too relevant here since this input isn't actually editable, but probably worth sticking to best practices so that in future copy-pastes all the existing examples are good!